### PR TITLE
[NO JIRA] Apply slug sanitization updates to taxonomies

### PR DESCRIPTION
**Pull request by apmatthews**
## Description

Accounting for the merge of #214:

- Replaces the `useApiIDGenerator` hook with `useInputGenerator`.
- Sanitizes taxonomy slugs server-side.
- Enforces slug length restrictions server-side.
- Changes API Identifier label to Taxonomy ID (following pattern from #200)
- Changes formatting of Taxonomy ID to match `sanitize_key()` behavior.
    - WordPress doesn't do this sanitization automatically on registration of taxonomies like they do for post types, so we wouldn't technically run into any mismatch between stored and registered taxonomy slug values. However, sanitizing still seems like a good thing for us to do. The taxonomy slug is used as an array key, so sanitize_key() still seemed like a good choice.

## Testing

Existing tests should provide sufficient coverage for these changes.

### CODE REVIEW

File: includes/rest-api/rest-api-endpoint-registration.php [hunk 1]
```diff
@@ -781,10 +781,13 @@ function content_model_multi_option_exists( array $names, string $current_choice
  * @since 0.6.0
  */
 function save_taxonomy( array $params, bool $is_update ) {
-	if ( empty( $params['slug'] ) ) {
+	// Sanitize key allows hyphens, but it's close enough to register_taxonomy() requirements.
+	$params['slug'] = isset( $params['slug'] ) ? sanitize_key( $params['slug'] ) : '';
+
+	if ( empty( $params['slug'] ) || strlen( $params['slug'] ) > 32 ) {
```
Comment (mindctrl) [inline includes/rest-api/rest-api-endpoint-registration.php:787]: ❤️ Love the strlen check

**Review by mindctrl — APPROVED** [review]

**Comment by apmatthews** [comment]
@mindctrl Just added 285c83c. Not a critical change, but is something I had planned to circle back to. Apologies for the re-ping!

**Review by nickcernis — APPROVED** [review]
Thanks for taking care of this!

File: includes/settings/js/src/components/Taxonomies.jsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ export default function Taxonomies() {
 		}
 
 		const getTaxonomyEditHref = (slug, type) =>
-			`edit-tags.php?taxonomy=${slug}&post_type=${type.toLowerCase()}`;
+			`edit-tags.php?taxonomy=${slug}&post_type=${type}`;
```
Comment (nickcernis) [inline includes/settings/js/src/components/Taxonomies.jsx:29]: Great addition, nice to be able to get rid of these.

**Pull request merged**